### PR TITLE
M2 all config options

### PR DESCRIPTION
**Pull request by l1mosh**

### CODE REVIEW

File: registration.php [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: limosh
+ * Date: 05/06/2018
+ * Time: 11:57
+ */
```
Comment (davidwindell) [inline registration.php]: Remove this

File: registration.php [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+\Magento\Framework\Component\ComponentRegistrar::register(
+    \Magento\Framework\Component\ComponentRegistrar::MODULE,
+    'Outeredge_ConfigProduct',
```
Comment (davidwindell) [inline registration.php]: Use UpperCamelCase `OuterEdge` here and throughout the module

File: view/frontend/requirejs-config.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
```
Comment (davidwindell) [inline view/frontend/requirejs-config.js]: Remove this statement

File: Plugin/Model/ResourceModel/Attribute/InStockOptionSelectBuilder.php [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+     * @param Status $stockStatusResource
+     */
+
+    public function __construct(Status $stockStatusResource, StockConfigurationInterface $stockConfiguration)
+    {
+        $this->stockStatusResource = $stockStatusResource;
+        $this->stockConfiguration = $stockConfiguration;
+    }
+
+
```
Comment (davidwindell) [inline Plugin/Model/ResourceModel/Attribute/InStockOptionSelectBuilder.php]: Remove empty line

File: Plugin/Model/ResourceModel/Attribute/InStockOptionSelectBuilder.php [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+     */
+    private $stockStatusResource;
+
+    private $stockConfiguration;
+
+
+    /**
+     * @param Status $stockStatusResource
+     */
+
```
Comment (davidwindell) [inline Plugin/Model/ResourceModel/Attribute/InStockOptionSelectBuilder.php]: Remove empty line

File: Helper/Data.php [hunk 6]
```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Outeredge\ConfigProduct\Helper;
+
+
```
Comment (davidwindell) [inline Helper/Data.php]: Remove empty line

File: Block/Product/View/Type/Configurable.php [hunk 7]
```diff
@@ -0,0 +1,206 @@
+<?php
+
+namespace Outeredge\ConfigProduct\Block\Product\View\Type;
+
+
```
Comment (davidwindell) [inline Block/Product/View/Type/Configurable.php]: Remove empty line

File: Block/Product/View/Type/Configurable.php [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+        return $stock;
+    }
+
+
+    /**
+     * Composes configuration for js
+     *
+     * @return string
+     */
+    public function getJsonConfig()
```
Comment (davidwindell) [inline Block/Product/View/Type/Configurable.php]: Is it necessary to override this function instead of operating on the parents return value?

**Review by davidwindell — CHANGES_REQUESTED** [review]

File: Block/Product/View/Type/Configurable.php (hunk 8 above)
Comment (l1mosh) [reply Block/Product/View/Type/Configurable.php]: regarding getJsonConfig(), i think that is the place to add the values in to the json. i need to have a look if i could achieve that any other way.

File: composer.json [hunk 9]
```diff
@@ -1,22 +1,18 @@
 {
     "name": "outeredge/edge-magento-module-allconfigurableoptions",
```
Comment (davidwindell) [inline composer.json]: Rename to `outeredge/magento-outofstockoptions-module`

Comment (l1mosh) [reply composer.json]: The rename changes only need to be in the composer.json rite?

**Review by davidwindell — APPROVED** [review]

**Pull request merged**